Sources/NetworkProtection/PacketTunnelProvider.swift

```diff
@@ -806,7 +810,9 @@ open class PacketTunnelProvider: NEPacketTunnelProvider {
             }
 
             Task {
-                try? await updateTunnelConfiguration(environment: settings.selectedEnvironment, serverSelectionMethod: serverSelectionMethod)
+                if case .connected = connectionStatus {
```

Comment:
We should not update the tunnel configuration if we're not connected, as it triggers a reconnect sort of event.  This could explain a lot of the toggle flickering we've been seeing.